portfolio/src/main/webapp/script.js

```diff
@@ -27,17 +27,18 @@ function addRandomRecommendation() {
   recommendationContainer.innerText = recommendation;
 }
 
-//gets comments from datastore
+/** get messages from datastore and display them */
 function getMessages() {
-  fetch('/messages').then(response => response.json()).then((comments) => {
+  fetch('/messages?num='+document.getElementById("num-comments").value+'&sort='+document.getElementById("sort-selection").value).then(response => response.json()).then((comments) => {
```

Comment:
Nitpick: it may be easier to read if you get the DOM elements outside the fetch function.